compiler/registry.go

```diff
@@ -24,6 +24,9 @@ func NewRegistry(fds *descriptorpb.FileDescriptorSet) (*Registry, error) {
 
 // FindExtensionByName implements protoregistry.ExtensionTypeResolver.
 func (f *Registry) FindExtensionByName(field protoreflect.FullName) (protoreflect.ExtensionType, error) {
+	if desc, err := protoregistry.GlobalTypes.FindExtensionByName(field); err == nil {
```

Comment:
Is this scoped correctly?

I have no idea  if it is, but typically scoping rules are innermost to outermost, but here it's searching globally then ... elsewhere?

Reply:
yup, this is scoped correctly - we are first looking for concrete types built in to the program and then dynamic types constructed from the filedescriptorset.